### PR TITLE
Add global exception handler and validation

### PR DESCRIPTION
**Pull request by vladshender**

### CODE REVIEW

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
There are changes from previous HW with mistakes.

**Comment by vladshender** [comment]
I do not understand what is the error?

**Comment by vladshender** [comment]
I created a separate branch, executed the task and made a request to the pool. The code changes are only visible for this task, I think. The problem is the organization of Git? I asked a question about Git in the chat, but I was not answered. Please help, what should I do?

File: src/main/java/com/example/dto/CreateBookRequestDto.java [hunk 1]
```diff
@@ -1,14 +1,30 @@
...
 import java.math.BigDecimal;
 import lombok.Data;
 
 @Data
 public class CreateBookRequestDto {
+    @NotNull
     private String title;
+
+    @NotNull
     private String author;
```
Comment (MentorMateAcademy) [inline src/main/java/com/example/dto/CreateBookRequestDto.java:15]: maybe better @notblank?

File: src/main/java/com/example/dto/UpdateBookRequestDto.java [hunk 2]
```diff
@@ -1,14 +1,30 @@
...
+
+    @NotNull
+    @DecimalMin(value = "10")
     private BigDecimal price;
+
+    @NotNull
     private String description;
+
+    @NotNull
     private String coverImage;
```
Comment (MentorMateAcademy) [inline src/main/java/com/example/dto/UpdateBookRequestDto.java:29]: What difference from CreateBookRequestDto?

File: src/main/java/com/example/exception/CustomGlobalExceptionHandler.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+            HttpStatusCode status,
+            WebRequest request
+    ) {
+        Map<String, Object> body = new LinkedHashMap<>();
+        body.put("timestamp", LocalDateTime.now());
+        body.put("status", HttpStatus.BAD_REQUEST);
+        List<String> errors = ex.getBindingResult().getAllErrors().stream()
+                .map(this::getErrorMessage)
+                .toList();
+        body.put("errors", errors);
```
Comment (MentorMateAcademy) [inline src/main/java/com/example/exception/CustomGlobalExceptionHandler.java:34]: save strings to constants

**Review by MentorMateAcademy — CHANGES_REQUESTED** [review]

**Pull request merged**